node/src/client/router.rs

```diff
@@ -179,24 +180,40 @@ impl<N: Network, C: ConsensusStorage<N>> Inbound<N> for Client<N, C> {
     fn block_request(&self, peer_ip: SocketAddr, message: BlockRequest) -> bool {
         let BlockRequest { start_height, end_height } = &message;
 
+        // Get the latest consensus version, i.e., the one for the last block's height.
+        let latest_consensus_version = match N::CONSENSUS_VERSION(end_height - 1) {
```

Comment:
Nit: Not sure if the `end_height` can ever be 0, but this would cause an underflow. A `saturating_sub` would be safer.

```suggestion
        let latest_consensus_version = match N::CONSENSUS_VERSION(end_height.saturating_sub(1)) {
```

Reply:
This PR also adds a check when deserializing `BlockRequest` that `start_height > 0` and `end_height > start_height`. This is to ensure that nodes can never request the genesis block or an empty range.

Should I still add the check?

Reply:
Would be nice, but it's a nit

Reply:
Done in [7c4b649](https://github.com/ProvableHQ/snarkOS/pull/3936/commits/7c4b6498bacb728a5bee29d82d54ad4214967e8f).